src/EVM.hs

```diff
@@ -2665,6 +2552,219 @@ writeMemory memory offset buf = do
     else
       pure memory
 
-freezeMemory :: MutableMemory s -> EVM s (Expr Buf)
+freezeMemory :: MutableMemory s -> EVM t s (Expr Buf)
 freezeMemory memory =
   ConcreteBuf . BS.pack . VUnboxed.toList <$> VUnboxed.freeze memory
+
+
+class VMOps (t :: VMType) where
```

Comment:
It would be nice to pull this out into it's own section at the top of the file (or maybe into `Types.hs`?)

Reply:
moved to Types, instances have to stay in EVM otherwise it would cause cyclic dependency